frontend/pages/SoftwarePage/SoftwareVersions/SoftwareVersions.tsx

```diff
@@ -1,11 +1,274 @@
-import React from "react";
+import React, { useCallback, useContext, useMemo } from "react";
+import { InjectedRouter } from "react-router";
+import { useQuery } from "react-query";
+import { Row } from "react-table";
+
+import PATHS from "router/paths";
+import { ISoftwareCountResponse } from "interfaces/software";
+import softwareAPI, {
+  ISoftwareCountQueryKey,
+  ISoftwareQueryKey,
+  ISoftwareTitlesResponse,
+  ISoftwareVersionsResponse,
+} from "services/entities/software";
+import { AppContext } from "context/app";
+import {
+  GITHUB_NEW_ISSUE_LINK,
+  VULNERABLE_DROPDOWN_OPTIONS,
+} from "utilities/constants";
+import { getNextLocationPath } from "utilities/helpers";
+
+// @ts-ignore
+import Dropdown from "components/forms/fields/Dropdown";
+import TableDataError from "components/DataError";
+import TableContainer from "components/TableContainer";
+import CustomLink from "components/CustomLink";
+import LastUpdatedText from "components/LastUpdatedText";
+import { ITableQueryData } from "components/TableContainer/TableContainer";
+
+import EmptySoftwareTable from "../components/EmptySoftwareTable";
+
+import generateSoftwareVersionsTableHeaders from "./SoftwareVersionsTableConfig";
 
 const baseClass = "software-versions";
 
-// interface ISoftwareVersionsProps {}
+interface IRowProps extends Row {
+  original: {
+    id?: number;
+  };
+}
+
+interface ISoftwareVersionsProps {
+  router: InjectedRouter;
+  isSoftwareEnabled: boolean;
+  query: string;
+  perPage: number;
+  orderDirection: "asc" | "desc";
+  orderKey: string;
+  showVulnerableSoftware: boolean;
+  currentPage: number;
+  teamId?: number;
+}
+
+const SoftwareVersions = ({
+  router,
+  isSoftwareEnabled,
+  query,
+  perPage,
+  orderDirection,
+  orderKey,
+  showVulnerableSoftware,
+  currentPage,
+  teamId,
+}: ISoftwareVersionsProps) => {
+  const { isSandboxMode, noSandboxHosts, isPremiumTier } = useContext(
+    AppContext
+  );
+
+  // request to get software versions data
+  const {
+    data: softwareVersionsData,
+    isLoading: isSoftwareVersionsLoading,
+    isError: isSoftwareVersionsError,
+  } = useQuery<
+    ISoftwareVersionsResponse,
+    Error,
+    ISoftwareVersionsResponse,
+    ISoftwareQueryKey[]
+  >(
+    [
+      {
+        scope: "software",
```

Comment:
Let's use difference scopes for `software_versions` and `software_titles`

Reply:
udpated